src/main/java/com/Elessar/app/util/MetricManager.java

```diff
@@ -0,0 +1,47 @@
+package com.Elessar.app.util;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+
+/**
+ * Created by Hans on 2/16/19.
+ */
+public class MetricManager {
+    private final Logger logger;
+    private final Map<String, Duration> operationDuration;
+    private final int maxEntryCount;
+    private final Lock lock;
+
+
+    public MetricManager(String name, int maxEntryCount) {
+        logger = LogManager.getLogger(name);
+        this.operationDuration = new HashMap<>();
+        this.maxEntryCount = maxEntryCount;
+        this.lock = new ReentrantLock();
+    }
+
+    public void add(String op, Duration duration) {
```

Comment:
I recommend change the signature to:
```
public void add(Metric m);
```
So if we later want to add more stuffs in Metric class the signature here doesn't need to change.